### DISCUSSION TITLE
[진짜급함]아래 UI 버튼 재 질문입니다.

### DISCUSSION THREAD

**Discussion by smartbh** [naverz/zepeto-studio-kor#62] [5. Q&A]
제페토 스튜디오에 있는 가이드를 보고 다시 만들어 봤습니다.

그러한데 좀 다른 부분이 있는거 같아 질문 드립니다.

![image](https://user-images.githubusercontent.com/43138813/145623742-1bc23ad6-b574-4ea1-99b5-21d0cd737e4f.png)

제페토 스튜디오에는 분명 이런식으로 설명이 되어 있습니다.

그래서 똑같이 버튼을 사용하기 위해 코드를 이런식으로 짰는데요

![image](https://user-images.githubusercontent.com/43138813/145623839-e7091791-e228-4026-9253-8f476c95d2f7.png)

이런식으로 코드를 짤 시 제페토 스튜디오에 올라온 예제와는 다르게

![image](https://user-images.githubusercontent.com/43138813/145623935-0bfb8ca4-3e3a-44ae-bbfa-8631ac97bd39.png)

인스펙터 창에서 btn 컴포넌트를 넣어 줄 수가 없습니다.

이 상태로 월드를 실행 시켜보면

![image](https://user-images.githubusercontent.com/43138813/145624050-6ab5c7d9-df25-4f9a-bc14-4ed25d6402ba.png)

일런식으로 onClick 프로퍼티를 찾지 못하였다는 오류가 뜹니다.

버튼 컴포넌트가 할당이 안되서 그런가 하여

혹시 private 접근 지정자가 문제라 그런가 싶어

public으로 접근 지정자를 바꿔주면

![image](https://user-images.githubusercontent.com/43138813/145624159-0259bb8a-7aba-41c3-81d8-1f231e3a38bf.png)

이런식으로 예제와 비슷하게 나옵니다.

그러나 버튼을 할당해 주고 시작후 버튼을 눌러봐도

다른 베이직 형태의 버튼처럼 눌리면서 회색으로 변하면서

버튼을 눌렀을시 나와야 하는

콘솔 로그창이 나오질 않습니다.

현재 버튼과 캔버스는

World Space에 이벤트 카메라는 제페토 카메라 입니다.

![image](https://user-images.githubusercontent.com/43138813/145625271-cfb9bb2f-5a93-4358-9925-924fe8e33c42.png)
![image](https://user-images.githubusercontent.com/43138813/145625296-3cd43533-a020-42d1-a764-0d97cc1ec5ae.png)

캔버스와 버튼 인스펙터 상태

**Comment by x4cf** [comment]
일단 퍼블릭으로 선언하고 버튼을 연결해주세요.
혹시 제페토 카메라는 어디서 가져와서 연결했을까요? 생성된 카메라가 아닌 페키지 속의 카메라가 연결된것으로 예상됩니다 :)

**Reply by smartbh** [reply]
헉 족집게시네요 Zepeto Players에 되어있던 Zepeto Camera를 짜라가서 찾아봤더니 패키지에 있기에 그걸 사용했었습니다.

그러면 동적으로 생성되는 로컬플레이어의 카메라를 가져와야한다는건데 방법을 모르겠네요; 이번에 유니티가 처음이라;
